xbmc/pvr/PVRChannelNumberInputHandler.cpp

```diff
@@ -80,9 +102,42 @@ void CPVRChannelNumberInputHandler::AppendChannelNumberCharacter(char cCharacter
   }
 
   if (m_inputBuffer.size() == static_cast<size_t>(m_iMaxDigits))
+  {
     m_inputBuffer.erase(m_inputBuffer.begin());
+    m_label = m_inputBuffer;
+  }
+  else if (m_inputBuffer.empty())
+  {
+    m_sortedChannelNumbers.clear();
+    GetChannelNumbers(m_sortedChannelNumbers);
+
+    // sort channel numbers alphabetically
+    std::sort(m_sortedChannelNumbers.begin(), m_sortedChannelNumbers.end());
```

Comment:
I have a feeling this fails since the numbers are strings, we would have to do a "natural sort"

Reply:
Also what about sub channel numbers?

Reply:
No, it works perfect. I tested it even with subchannel numbers.

Reply:
<img width="543" alt="screen shot 2018-04-25 at 20 45 42" src="https://user-images.githubusercontent.com/3226626/39266251-c58ccae6-48c9-11e8-91dc-ea01801fe15b.png">

This is how it looks after std:.sort